packages/@n8n/nodes-langchain/nodes/llms/LMChatOpenAi/LmChatOpenAi.node.ts

```diff
@@ -186,6 +189,203 @@ export class LmChatOpenAi implements INodeType {
+						name: 'mcpServers',
+						type: 'fixedCollection',
+						placeholder: 'Add MCP Server',
+						typeOptions: {
+							multipleValues: true,
+							sortable: true,
+						},
+						default: { mcpServerOptions: [{ serverLabel: '' }] },
+						options: [
+							{
+								displayName: 'MCP Server',
+								name: 'mcpServerOptions',
+								values: [
+									{
+										displayName: 'Server Label',
+										name: 'serverLabel',
+										type: 'string',
+										default: '',
+										description: 'A label to identify this MCP server',
+										placeholder: 'e.g. My Database Server',
+									},
+									{
+										displayName: 'Connection Type',
+										name: 'connectionType',
+										type: 'options',
+										default: 'url',
+										options: [
+											{ name: 'URL', value: 'url' },
+											{ name: 'Connector ID', value: 'connector_id' },
+										],
+										description: 'Choose how to connect to the MCP server',
+									},
+									{
+										displayName: 'Server URL',
+										name: 'serverUrl',
+										type: 'string',
+										default: '',
+										description: 'The URL of the MCP server',
+										placeholder: 'e.g. https://api.example.com/mcp',
+										displayOptions: {
+											show: {
+												connectionType: ['url'],
+											},
+										},
+									},
+									{
+										displayName: 'Connector ID',
+										name: 'connectorId',
+										type: 'string',
+										default: '',
+										description: 'The connector ID for the MCP server',
+										placeholder: 'e.g. connector_gmail',
+										displayOptions: {
+											show: {
+												connectionType: ['connector_id'],
+											},
+										},
+									},
+									{
+										displayName: 'Authorization',
+										name: 'authorization',
+										type: 'string',
+										default: '',
+										description: 'Authorization token or credentials for the MCP server',
+									},
+									{
+										displayName: 'Headers',
+										name: 'headers',
+										type: 'json',
+										default: '{}',
+										description: 'Additional headers to send with requests to the MCP server',
+										placeholder: '{"X-Custom-Header": "value"}',
+									},
+									{
+										displayName: 'Server Description',
+										name: 'serverDescription',
+										type: 'string',
+										default: '',
+										description: 'A description of what this MCP server provides',
+										placeholder: 'e.g. Database access for user management',
+										typeOptions: {
+											rows: 2,
+										},
+									},
+									{
+										displayName: 'Allowed Tools',
+										name: 'allowedTools',
+										type: 'string',
+										default: '',
+										description:
+											'Comma-separated list of tools to allow. If not provided, all tools will be allowed.',
+										placeholder: 'e.g. tool1,tool2',
+									},
+								],
+							},
+						],
+					},
```

Comment:
What's the use-case for this? Why wouldn't user just use our native MCP client node which has more robust authentication support and is easier to debug.